internal/pkg/service/common/configmap/flags.go

```diff
@@ -41,8 +47,8 @@ func GenerateFlags(fs *pflag.FlagSet, v any) error {
 				return nil
 			}
 
-			shorthand := vc.StructField.Tag.Get(configShorthandTag)
-			usage := vc.StructField.Tag.Get(configUsageTag)
+			shorthand := vc.Shorthand
+			usage := vc.Usage
```

Comment:
The main part of the bug. In the `configmap.Value` use-case, we have to use parent metadata.